src/CustomMarshalers/src/System/Runtime/InteropServices/CustomMarshalers/EnumeratorViewOfEnumVariant.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+
+namespace System.Runtime.InteropServices.CustomMarshalers
+{
+    internal class EnumeratorViewOfEnumVariant : ICustomAdapter, IEnumerator
+    {
+        private readonly IEnumVARIANT _enumVariantObject;
+        private bool _fetchedLastObject;
+        private object[] _nextArray = new object[1];
+
+        public EnumeratorViewOfEnumVariant(IEnumVARIANT enumVariantObject)
+        {
+            _enumVariantObject = enumVariantObject;
+            _fetchedLastObject = false;
+            Current = null;
+        }
+
+        public object Current { get; private set; }
```

Comment:
Nit: We tend to avoid mixing of implicit properties and explicit state. It makes it hard to see the full state of the object; and generates a tiny bit slower/bigger code too.